### PR TITLE
improve discussion in gplvm tutorial

### PR DESCRIPTION
**Pull request by martinjankowiak**
i may need to make scrub this but i can't get it to work on my system

### CODE REVIEW

File: tutorial/source/gplvm.ipynb [hunk 1]
```diff
@@ -315,63 +319,67 @@
...
-   "execution_count": 4,
+   "execution_count": 15,
    "metadata": {},
    "outputs": [],
    "source": [
     "captured_time = y.new_tensor([int(cell_name.split(\" \")[0]) for cell_name in df.index.values])\n",
-    "# we scale the time info into the interval [0, 1]\n",
-    "time = (captured_time.log2() - 1) / 5\n",
+    "# we scale the time into the interval [0, 1]\n",
+    "time = captured_time.log2() / 6\n",
```
Comment (martinjankowiak) [inline tutorial/source/gplvm.ipynb]: i changed this to get it into the unit interval

File: tutorial/source/gplvm.ipynb [hunk 2]
```diff
@@ -297,13 +299,15 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "First, we need to define the output $y$. To predict values of all $48$ genes, we need $48$ Gaussian processes. So in Pyro GP, the required shape for $y$ is `num_GPs x num_data = 48 x 437`."
+    "First, we need to define the output tensor $y$. To predict values for all $48$ genes, we need $48$ Gaussian processes. So the required shape for $y$ is `num_GPs x num_data = 48 x 437`."
```
Comment (martinjankowiak) [inline tutorial/source/gplvm.ipynb]: what do the `y` observations actually represent?

File: tutorial/source/gplvm.ipynb [hunk 3]
```diff
@@ -315,63 +319,67 @@
...
+   "metadata": {
+    "collapsed": true
+   },
    "outputs": [],
    "source": [
     "kernel = gp.kernels.RBF(input_dim=2, lengthscale=torch.ones(2))\n",
     "\n",
-    "# we clone to avoid the change of value of X_init during the inference\n",
-    "X = Parameter(X_init.clone())\n",
+    "# we clone here so that we don't change our prior during the course of training\n",
```
Comment (martinjankowiak) [inline tutorial/source/gplvm.ipynb]: is this correct?

File: tutorial/source/gplvm.ipynb [hunk 4]
```diff
@@ -386,45 +394,47 @@
...
...9JM3vOzB4Jr+P9eXR3PQZ5EF3S5RXg\nTKAeeAGYU+UaXgemZrX9D2BlGF4J3ByG54QaG4DZofZkBWuZD1wAvFxOLcDTwDyim8k9Cnx8GOr6\nBvBfcoxblbqAM4ALwvAE4Pdh3rEurzx1xb28DBgfhuuAdaHvuJfXYHXFurwy5vdl4EfAI7Xw86g1\nlPxq9SZeS4BVYXgVcGVG+/3u3u3urwEdRJ+hItz9V0BXObWY2RnARHd/yqNv8z0Z01SyrsFUpS53\n3+Huz4bhg8Amovv1xLq88tQ1mGrV5e5+KLysCw8n/uU1WF2Dqdr33sxmAFcAd2bNP7blpUDJrxZu\n4uXAE2a23sxWhLZp7r4jDO8EpoXhOOottpbpYbgaNX7ezF4Mm8TSq/5Vr8vMZgHnE/11WzPLK6su\niHl5hc03zwO7gTZ3r4nlNUhdEP/363vAV4D+jLZYl5cCpfZd4u7nAR8HrjWz+Zlvhr8qauJQvVqq\nBbidaFPlecAO4DtxFGFm44GfAF909wOZ78W5vHLUFfvycve+8F2fQfTX87lZ78eyvAapK9blZWaf\nBHa7+/rBxoljeSlQ8ivoJl7Dyd23h+fdwE+JNmHtCquqhOfdYfQ46i22lu1heFhrdPdd4RdBP/BP\nHN/0V7W6zKyO6Jf2fe7+UGiOfXnlqqsWlleau+8DfgEspgaWV666amB5XQx8ysxeJ9oUf5mZ/ZCY\nl5cCJb9Yb+JlZuPMbEJ6GFgIvBxqWBZGWwY8HIbXAEvNrMHMZgMtRDvchlNRtYTV8QNmNi8cTXJ1\nxjQVk/6hCv4t0XKrWl2hj7uATe7+3Yy3Yl1eg9VVA8ur2cxOC8NjgcuB3xH/8spZV9zLy92vc/cZ\n7j6L6PfSk+7+GeL+eSx1b/5oeQCfIDoS5hXga1We95lER2a8AGxIzx+YAqwFtgBPAE0Z03wt1LqZ\nChxFklXPj4lW73uItrUuL6UWoJXoB/AV4H8RTrCtcF33Ai8BL4YfpjOqWRdwCdHmhheB58PjE3Ev\nrzx1xb283g88F+b/MvD1Ur/rVaor1uWVVeMfc/wor1iXl86UFxGRitAmLxERqQgFioiIVIQCRURE\nKkKBIiIiFaFAERGRilCgiIhIRShQRESkIhQoIiJSEf8f8dztHtZcYYgAAAAASUVORK5CYII=\n",
```
Comment (martinjankowiak) [inline tutorial/source/gplvm.ipynb]: dunno why the plot now has a green line?

File: tutorial/source/gplvm.ipynb (hunk 3 above)
Comment (fehiepsi) [reply tutorial/source/gplvm.ipynb]: yes, it is correct! It seems that I cloned `X_init` in the prior too, which is unnecessary.

File: tutorial/source/gplvm.ipynb (hunk 1 above)
Comment (fehiepsi) [reply tutorial/source/gplvm.ipynb]: you are right! I can't believe that I made that mistake. Thanks!

File: tutorial/source/gplvm.ipynb (hunk 2 above)
Comment (fehiepsi) [reply tutorial/source/gplvm.ipynb]: `y` is the high dimensional output, which is what we want to do dimensional reduction on.

Comment (martinjankowiak) [reply tutorial/source/gplvm.ipynb]: i'm not talking about math... i mean the thing that's done in the lab

File: tutorial/source/gplvm.ipynb (hunk 4 above)
Comment (fehiepsi) [reply tutorial/source/gplvm.ipynb]: I fetch your branch. It does not show the green line in my system. I suspect your `matplotlib` did some magical thing here. :)

File: tutorial/source/gplvm.ipynb [hunk 5]
```diff
@@ -315,63 +319,67 @@
...
-   "metadata": {},
+   "execution_count": 17,
+   "metadata": {
+    "collapsed": true
+   },
    "outputs": [],
    "source": [
     "# we use `.to_event()` to tell Pyro that the prior distribution for X has no batch_shape\n",
-    "gplvm.set_prior(\"X\", dist.Normal(X_init.clone(), 0.1).to_event())\n",
+    "gplvm.set_prior(\"X\", dist.Normal(X_prior_mean.clone(), 0.1).to_event())\n",
```
Comment (fehiepsi) [inline tutorial/source/gplvm.ipynb]: nit: no need for `clone` here

File: tutorial/source/gplvm.ipynb [hunk 6]
```diff
@@ -386,45 +394,47 @@
...
      "output_type": "display_data"
     }
    ],
    "source": [
+    "# note that training is expected to take a minute or so\n",
     "losses = gp.util.train(gplvm, num_steps=4000)\n",
     "\n",
     "# let's plot the loss curve after 4000 steps of training\n",
-    "plt.plot(losses);"
+    "plt.plot(losses);\n",
```
Comment (fehiepsi) [inline tutorial/source/gplvm.ipynb]: nit: no need for the semicolon because we already have `plt.show()`

File: tutorial/source/gplvm.ipynb [hunk 7]
```diff
@@ -315,63 +319,67 @@
...
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 15,
    "metadata": {},
    "outputs": [],
    "source": [
     "captured_time = y.new_tensor([int(cell_name.split(\" \")[0]) for cell_name in df.index.values])\n",
```
Comment (fehiepsi) [inline tutorial/source/gplvm.ipynb]: nit: sorry, can we change this to "capture_time"?

**Comment by fehiepsi** [comment]
About `make scrub` and matplotlib issue, if you are not going to make any change, I will do `make scrub`, rerun the notebook, and push into your branch (I am not sure whether I can push; if not, I'll make another PR).

**Pull request merged**